### PR TITLE
added io ops

### PR DESCRIPTION
**Pull request by aujjwal-redhat**
Added the following ops:

* create_link_file

Fixes: #784 

Signed-off-by: Ayush Ujjwal <aujjwal@redhat.com>

### CODE REVIEW

File: common/ops/support_ops/io_ops.py [hunk 1]
```diff
@@ -1169,3 +1169,76 @@ def list_files(self, node: str, dir_path: str, parse_str: str = "",
                               f"{dir_path} on node {node} for user {user}")
             return None
         return ret['msg']
+
+    def git_clone_and_compile(self, nodes: str, link: str, dir_name: str,
```
Comment (srijan-sivakumar) [inline common/ops/support_ops/io_ops.py:1173]: @aujjwal-redhat this won't work in most of the cases. If a test case needs it, it should be added in the TC. The compilation steps will vary for the repo being cloned and the framework or language being used in it.

Comment (aujjwal-redhat) [reply common/ops/support_ops/io_ops.py:1173]: Ack

**Comment by srijan-sivakumar** [comment]
@aujjwal-redhat can you add an example in the test_io_ops.py ?

**Comment by aujjwal-redhat** [comment]
> @aujjwal-redhat can you add an example in the test_io_ops.py ?

Done

**Review by srijan-sivakumar — APPROVED** [review]

**Pull request merged**